app/modules/pipe.c

```diff
@@ -3,22 +3,54 @@
 ** table to store the LUAL_BUFFERSIZE byte array chunks instead of the stack.
 ** Writing is always to the last UD in the table and overflow pushes a new UD to
 ** the end of the table.  Reading is always from the first UD in the table and
-** underrun removes the first UD to shift a new one into slot 1.
+** underrun removes the first UD to shift a new one into slot 2. (Slot 1 of the
+** table is reserved for the pipe reader function with 0 denoting no reader.)
 **
 ** Reads and writes may span multiple UD buffers and if the read spans multiple UDs
 ** then the parts are collected as strings on the Lua stack and then concatenated
-** with a `lua_concat()`.
+** with a lua_concat().
 **
-** Note that pipes also support the undocumented length and tostring operators
-** for debugging puposes, so if p is a pipe then #p[1] gives the effective
-** length of pipe slot 1 and printing p[1] gives its contents
+** Note that pipe tables also support the undocumented length and tostring
```

Comment:
Looks like it's no longer undocumented! ;)

Reply:
Documented as in readthedocs general documentation.  Anyone reading this is digging around in the pipe internals anyway.